### ISSUE TITLE
How to avoid the scroll of the browser window when clicking on OSD canvas?

### ISSUE THREAD

**Issue by szenzaro** [openseadragon/openseadragon#2071]
Hello, I would like to configure OSD in order to avoid the scroll of the browser window the first time you click on OSD instance.
It is possible to verify the behaviour even in OSD web site https://openseadragon.github.io/ 

If you click on the "OpenSeadragon Viewer With Default Settings" the page will scroll to bring up OSD.

Is there a configuration option or a way to avoid this behaviour?
Is it the desired behaviour?

**Comment by iangilman** [comment]
So you're on the page but the viewer is only partially visible and you click on it and the page scrolls so that it's completely visible? I think this is a side effect of giving the viewer keyboard focus (for https://openseadragon.github.io/examples/ui-keyboard-navigation/); the browser automatically scrolls the focused element into view. You could potentially disable it by making OSD not focus when you click. There's currently no way to do that through the API, but you could make a new build with this line commented out: 

https://github.com/openseadragon/openseadragon/blob/c8f4934c8a5f33b15830d9a82e5443a48519db63/src/viewer.js#L2839

Of course it would make it so your users couldn't use keyboard commands to navigate the viewer, but maybe that's not important in your case. 

I'm curious what your scenario is where this is undesired behavior. I do think in most scenarios it's desirable to have the viewer scroll into view when you start to interact with it, but I'm sure there are cases where one wouldn't want that!

**Comment by blms** [comment]
Hello!

I am working on a project where we are using OSD with some IIIF images that may be taller than the viewport. We also have our own controls above OSD, replacing OSD's native control panel overlay. We are encountering an issue, particularly on mobile, where OSD scrolls itself into view on first interaction and, if the image is taller than the viewport, makes our custom controls inaccessible by scrolling past them. This is worse on mobile because it creates a scroll trap; we have OSD taking up 100% of the screen width, but we still need mobile users to be able to access the controls above OSD after interacting with it.

As a workaround, we truncate the OSD container's height with CSS to be `100vw` minus the height of our controls. (By "OSD container" I mean OSD's parent element set via the `element` option.) That way, at least when the bottom of OSD is outside of the viewport at first, the browser only scrolls far enough that our controls are still visible. But that's a bit hacky; for our use case, it would certainly be useful to have a setting to turn this off instead!

**Comment by iangilman** [comment]
@blms Yeah, this is a perfect case for not supporting keyboard focus, since it's on mobile. As I mentioned above, there's no way in the API currently to turn it off, but it shouldn't be too hard to modify the OSD code to support it. Would you be interested in pursuing a patch for this?

**Comment by rlskoeser** [comment]
@iangilman how much effort do you think it would be to submit a patch? what would be involved, adding & documenting a new configuration and honoring it? how hard to write a unit test for this behavior, do you think?

(FYI @blms is working with me on the project where this came up)

**Comment by iangilman** [comment]
Shouldn't be too hard... Seems like a pretty simple feature. But yes, we would want a new config flag (unless we want to try to automatically detect mobile, but that seems like it might be problematic at the library level) and it would need some documentation comments. We're actually not too hardcore about adding tests for every new feature, though a test would certainly be welcome! 

Here are our contribution guidelines: 

https://github.com/openseadragon/openseadragon/blob/master/CONTRIBUTING.md

My recommendation is to go ahead and get started on a fork and see if it's useful to you. From there, you already have something that helps you, but also it shouldn't be too much further to a PR that can help everyone :)

**Comment by rlskoeser** [comment]
Thanks! Absolutely agree, should be configured and not attempt to auto-detect. Will discuss with @blms next week.

**Comment by iangilman** [comment]
Sounds great!
